### ISSUE TITLE
Add API for gamepads and joysticks

### ISSUE THREAD

**Issue by Jacalz** [fyne-io/fyne#4454]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

While Fyne isn't designed to be used for building games necessarily, there are other use-cases where it makes sense to have a clean and simple interface for accessing gamepads and joysticks. As an example, we are building a control application (not built in Fyne but might as well have been) for a remote controlled car and it would be nice to steer with the controller when not in automatic driving mode.

### Is it possible to construct a solution with the existing API?

GLFW has some kind of support for joysticks and gamepads, see https://www.glfw.org/docs/latest/input_guide.html#joystick.

### Describe the solution you'd like to see.

Some kind of coherent API, not necessarily too different from the keyboard button API, for accessing gamepads and joysticks.

**Comment by andydotxyz** [comment]
It's interesting to consider how remotes overlap with this. There is an open PR to add some android remote buttons and I wonder if it fits closer to gamepad than keyboard?

**Comment by Jacalz** [comment]
Yeah, I think that might be the case. I suspect that introducing gamepad buttons into the keyboard details would make us run into problems when we want to add joystick support (well, axis controls in general) later on down the line.
